plugins/woocommerce-admin/client/products/sections/images-section.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Link,
+	useFormContext,
+	MediaUploader,
+	ImageGallery,
+	ImageGalleryItem,
+} from '@woocommerce/components';
+import { Card, CardBody, DropZone } from '@wordpress/components';
+import { recordEvent } from '@woocommerce/tracks';
+import { useState } from '@wordpress/element';
+import { Product } from '@woocommerce/data';
+import classnames from 'classnames';
+import { Icon, trash } from '@wordpress/icons';
+import { MediaItem } from '@wordpress/media-utils';
+
+/**
+ * Internal dependencies
+ */
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import DragAndDrop from '../images/drag-and-drop.svg';
+import './images-section.scss';
+
+type Image = MediaItem & {
+	src: string;
+};
+
+export const ImagesSection: React.FC = () => {
+	const { getInputProps, setValue } = useFormContext< Product >();
+	const images = ( getInputProps( 'images' ).value as Image[] ) || [];
+	const [ showRemoveZone, setShowRemoveZone ] = useState( false );
+	const [ imageIdToRemove, setImageIdToRemove ] = useState< number >( 0 );
+
+	const getUniqueImages = ( files: Image[] ) => {
+		if ( ! files ) {
+			return [];
+		}
+		files.forEach( ( image: Image ) => {
+			if (
+				image.id &&
+				images.find( ( file ) => file.id === image.id ) === undefined
+			) {
+				images.push( image );
+			}
+		} );
+		setValue( 'images', images );
```

Comment:
This `getUniqueImages` function has a side effect of `setValue` which feels out of place.  We're also calling `setValue` below again after calling `getUniqueImages`.

```
								setValue(
									'images',
									getUniqueImages( [ file ] )
								)
```

Reply:
Nice catch! I removed the `setValue( 'images', images );` inside the function in commit [5ceceaa](https://github.com/woocommerce/woocommerce/pull/34769/commits/5ceceaadcb559334ea757434b6780e7380c658ed)